src/Processors/QueryPlan/ReadFromMergeTree.cpp

```diff
@@ -1443,7 +1442,10 @@ MergeTreeDataSelectAnalysisResultPtr ReadFromMergeTree::selectRangesToReadImpl(
     const Names & primary_key_column_names = primary_key.column_names;
 
     if (!indexes)
-        buildIndexes(indexes, query_info.filter_actions_dag, data, context, query_info, metadata_snapshot);
+        buildIndexes(indexes, query_info.filter_actions_dag, data, parts, context, query_info, metadata_snapshot);
+
+    if (indexes->part_values && indexes->part_values->empty())
+        return std::make_shared<MergeTreeDataSelectAnalysisResult>(MergeTreeDataSelectAnalysisResult{.result = std::move(result)});
```

Comment:
This breaks filtering partitions for non-analyzer (should be fixed by #54449) and also it does not work even for analyzer without #5448? (note that it got reverted)

Reply:
Also I've tried to re-apply #52762, but newly added test in #54449 did not pass either